### PR TITLE
feat(series): nft top collection

### PR DESCRIPTION
**Pull request by roiLeo**
⚠️ WORK IN PROGRESS ⚠️ 
Some of the code is dirty (Rankings/utils.ts) and need refatoring/rework.
I'm not sure if I got the right 7d% & 30d% formula (need some verification).

### Brief
New page/component Rankings

### PR type
- [x] Feature

### Before submitting this PR, please make sure:
- [ ] Code builds clean without any errors or warnings
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've didn't break any original functionality
- [x] I've posted screenshot of demonstrated change in this PR

### Optional
- [ ] I've tested it on mobile and everything responsive doesn't work on image

### Need to test
- [ ] verification of volume price
- [ ] verification of floor price
- [ ] verification of 7d%/30d% data

### Rework
- [ ] 7d%/30d% Formula: 
As if t_0 (price now) & t_7 (price 7d ago) so: 
`price_change(%) = ((price(t_0) - price(t_24)) / price(t_0)) * 100`

### What's new? (may be part of changelog)
- #586 related
- #587 

### Screenshot
Screenshot 2021-09-30 at 11-35-31 KodaDot Kusama NFT Market explorer

### CODE REVIEW

**Comment by yangwao** [comment]
Added right issue to desc https://github.com/kodadot/nft-gallery/issues/587 however #586 was just initial draft 😄

File: src/components/Rankings/RankingsTable.vue [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+	@Prop() public value!: any
+	protected data: RowRanking[] = []
+	protected columns: Column[] = columns
+	protected usersWithIdentity: RowRanking[] = []
+	protected nbRows: number = 10
+
+	public meta: NFTMetadata = emptyObject<NFTMetadata>()
+
+	async created() {
+		await this.fetchCollectionsRankings()
```
[the remark this answers is not in the document]
Comment (roiLeo) [reply src/components/Rankings/RankingsTable.vue:222]: same speed with and without ~3800ms

File: src/components/Rankings/RankingsTable.vue [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+  }
+
+  @Watch('nbRows')
+  public onTopRowsChange(value: string) {
+    console.log(value)
+    this.fetchCollectionsRankings(Number(value))
+  }
+
+	public async fetchMetadataImage(metadata: any) {
+		const meta = await fetchCollectionMetadata({ metadata } as Collection)
```
[the remark this answers is not in the document]
Comment (roiLeo) [reply src/components/Rankings/RankingsTable.vue:266]: wdym by caching collecion metadata ? like in CollectionItem ?

[the remark this answers is not in the document]
Comment (vikiival) [reply src/components/Rankings/RankingsTable.vue:266]: like use `idb` for persisting collection metadata

[the remark this answers is not in the document]
Comment (yangwao) [reply src/components/Rankings/RankingsTable.vue]: > let's cache Collection metadata

how invalidation would work not showing user super old data?

**Comment by yangwao** [comment]
Should we start testing it? 👀
This will be rocket fuel for KodaDot 🚨

**Comment by roiLeo** [comment]
> Should we start testing it? 👀 This will be rocket fuel for KodaDot 🚨

Well, problem is %7d & %30d stats seems wrong as they will always be green or null.
Currently to calculate percentage over 7 days, I compare traded volume over 7 days on total volume, this will never be negative.

**Comment by yangwao** [comment]
I would add addressable rows to address-bar, ?rows=100 ? :)
image

**Comment by yangwao** [comment]
I was thinking if we could make it more asynchronously, so each row would be own query to have it feel less blocking on frontend?

Or Viki proposed just fetch name of collections and then fetch data for collections as separate query?

**Comment by roiLeo** [comment]
> I was thinking if we could make it more asynchronously, so each row would be own query to have it feel less blocking on frontend?
> 
> Or Viki proposed just fetch name of collections and then fetch data for collections as separate query?

I don't think this will resolve loading time. IMO, problem comes from the query that returns all the data, to reduce it we could sort by rank directly from graphql query and add pagination (first & offset).

**Comment by yangwao** [comment]
Let's experiment with that :)
Anything that could help put loading times down 👌

**Comment by yangwao** [comment]
Oh re: naming
We've been chatting about the naming of this feature. 
We've agreed that we would go with  
**SERIES** on nav and URL could be /series-insights

**Comment by yangwao** [comment]
Checking on verifications, is that okayish or still needs to be checked?

> Need to test
> - verification of volume price
> - verification of floor price
> - verification of 7d%/30d% data

Checking what's left important from issue https://github.com/kodadot/nft-gallery/issues/587

Would be good to have 
- Number of different owners

Then is ready to fly, for **silent** update for now, so ppl can verify it is playing all right 👌 😄

**Comment by yangwao** [comment]
LGTM

**Pull request merged**